### PR TITLE
tests: update nested wait for snapd command

### PR DESCRIPTION
**Pull request by sergiocazzolato**
I see that sometimes the uc20 nested tests fail because of a disconnection
with the remote vm. The problem seems to be a race because the snpad
snap is installed while the tools are being prepared.

To fix that the idea is to prepare the tools after we check snap command
is ready.

error in test:
2021-07-30T02:21:20.4738952Z Warning: Permanently added
'[localhost]:8022' (ECDSA) to the list of known hosts.
2021-07-30T02:21:20.4740233Z
PATH=/writable/test-tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
2021-07-30T02:21:20.4741638Z Warning: Permanently added
'[localhost]:8022' (ECDSA) to the list of known hosts.
2021-07-30T02:21:20.4742645Z Connection to localhost closed by remote
host.
2021-07-30T02:21:20.4743582Z -----

vm journal log:
[   23.550890] systemd-logind[878]: New session 21 of user user1.
[   23.554706] systemd[1]: Started Session 21 of user user1.
[   24.088541] systemd[1]: Created slice system-console\x2dconf.slice.
[   24.095905] systemd[1]: Created slice
system-serial\x2dconsole\x2dconf.slice.
[   24.318152] sshd[1527]: Connection closed by 10.0.2.2 port 50750
[preauth]
[   24.462615] systemd[1]: Reloading.
[   24.674621] systemd[1]: emergency.target: Requested dependency
OnFailure=reboot.target ignored (target units cannot fail).
[   24.869609] systemd[1]: Reloading.
[   25.070768] systemd[1]: emergency.target: Requested dependency
OnFailure=reboot.target ignored (target units cannot fail).
[   25.238908] systemd[1]: Mounting Mount unit for snapd, revision x1...
[   25.299687] systemd[1]: Mounted Mount unit for snapd, revision x1.
[   25.763110] systemd[1]: Reloading.
[   25.991015] systemd[1]: emergency.target: Requested dependency
OnFailure=reboot.target ignored (target units cannot fail).
[   26.164923] systemd[1]: Reloading.

### CODE REVIEW

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -52,7 +52,19 @@ nested_wait_for_no_ssh() {
 }
 
 nested_wait_for_snap_command() {
-    nested_exec "retry --wait 1 -n 200 sh -c 'command -v snap'"
+    # In this function cannot be used the remote retry command because it could
```
Comment (mardy) [inline tests/lib/nested.sh:55]: nitpick: "In this function the remote retry command cannot be used because..."

Comment (mardy) [reply tests/lib/nested.sh:55]: Do I understand correctly, that the problem is that the `retry` command itself might not be there at this time?

Comment (sergiocazzolato) [reply tests/lib/nested.sh:55]: Right, when we use nested_wait_for_snap_command after creating/booting a new image, the retry command is not yet deployed.

Comment (sergiocazzolato) [reply tests/lib/nested.sh]: Comment updated, thanks for review

**Review by mvo5 — APPROVED** [review]
Thank you

**Review by mardy — APPROVED** [review]
Thanks!

**Pull request merged**